gtdynamics/universal_robot/FixedJoint.h

```diff
@@ -0,0 +1,57 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  FixedJoint.h
+ * @brief Representation of a fixed joint.
+ * @author Varun Agrawal
+ */
+
+#pragma once
+
+#include "gtdynamics/universal_robot/Joint.h"
+
+namespace gtdynamics {
+
+/**
+ * @class FixedJoint is a joint which has 0 degrees of freedom.
+ */
+class FixedJoint : public Joint {
+ private:
+  JointParams fixedJointParams() {
+    JointParams params;
+    params.effort_type = JointEffortType::Unactuated;
+    params.velocity_limit = 0.0;
+    params.acceleration_limit = 0.0;
+    params.torque_limit = 0.0;
+    return params;
+  }
+
+ public:
+  /**
+   * @brief Create FixedJoint using joint name, joint pose in
+   * base frame, and parent and child links.
+   *
+   * Since a fixed joint has 0 degrees of freedom, we set the screw axis to
+   * Vector6::Zero.
```

Comment:
does this ever have ILS issues due to joint angle having no contribution?

Reply:
It shouldn't but I'm not sure if a unit test actually tests for this.